controllers/core/openstackversion_controller.go

```diff
@@ -255,7 +255,7 @@ func (r *OpenStackVersionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			corev1beta1.OpenStackVersionMinorUpdateReadyMessage)
 
 		// minor update for Controlplane in progress
-		if !openstack.ControlplaneContainerImageMatch(controlPlane, instance) ||
+		if !openstack.ControlplaneContainerImageMatch(ctx, controlPlane, instance) ||
```

Comment:
could we improve the condition here to reflect the components where there is a missmatch?
like when ControlplaneContainerImageMatch returns a list of strings with the components where there is a miss match we could list it in the condition message.

also if the component IsReady, but there is a missmatch, it would be no longer a minor update is running, it would be an minor update error?

Reply:
Just added a commit which adds this functionality to this PR